### HYPOTHESIS TESTING FORMULATION

HYPOTHESIS FORMULATION 

Load relevant data and csv files :
- The dataset has no missing or duplicate values.
- Numeric columns like Revenue and Budget are in the correct formats 

For our two hypothesis that seek to explain the `financial projections for our investors` below:
1.  Genre comparison ttest:Analyse Action,Adventure,scifi vs. Adventure,Animation,Comedy

2.  Budget  comparison ttest:Analyse High-budget ($>$ $17M) vs. low-budget ($\leq$ $17M).

let’s establish the hypotheses:

A.` analysis of genres investigates whether the average production budgets differ significantly between Action,adventure and sci-Fi and Adventure ,Animation,Comedy films. A two-sample t-test was used to assess the difference `
- Null Hypothesis (H₀): The average box office revenue for Action/Adventure/Sci-Fi movies is equal to that of Adventure/Animation/Comedy movies.
- Alternative Hypothesis (H₁): The average box office revenue for Action/Adventure/Sci-Fi movies is different (or higher) than Adventure/Animation/Comedy movies.

B.`Impact of Budget on Box Office Revenue
To evaluate whether investments on production budgets significantly affect box office performance.A two-sample t-test was conducted comparing revenues from`
- Null Hypothesis (H₀): Budget has no significant impact on box office revenue.
- Alternative Hypothesis (H₁): Higher budgets lead to significantly higher box office revenue




In [87]:
#import necessary libraries
import pandas as pd 
import numpy as np

#### Loading the all combined csv to our `Film_df`

In [88]:
film_df = pd.read_csv("zippedData/all_combined.csv")
film_df

c:\Users\hp\anaconda3\envs\learn-env\lib\site-packages\IPython\core\interactiveshell.py:3145: DtypeWarning: Columns (19) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


movie_id  ordering   person_id         category  ordering.1  \
0      tt0475290        10   nm0005683  cinematographer          24   
1      tt0475290         1   nm0000982            actor          24   
2      tt0475290         2   nm0000123            actor          24   
3      tt0475290         3   nm2403277            actor          24   
4      tt0475290         4   nm0000146            actor          24   
...          ...       ...         ...              ...         ...   
50962  tt7153766         8   nm2722502         producer           8   
50963  tt7153766         8   nm2722502         producer          15   
50964  tt7153766         9  nm10426133         composer           7   
50965  tt7153766         9  nm10426133         composer           8   
50966  tt7153766         9  nm10426133         composer          15   

               title region  is_original_title  primary_title  \
0      Hail, Caesar!     GB                0.0  Hail, Caesar!   
1      Hail, Caesar!     GB                0.0  Hail, Caesar!   
2      Hail, Caesar!     GB                0.0  Hail, Caesar!   
3      Hail, Caesar!     GB                0.0  Hail, Caesar!   
4      Hail, Caesar!     GB                0.0  Hail, Caesar!   
...              ...    ...                ...            ...   
50962         Unsane     IT                0.0         Unsane   
50963         Unsane     US                0.0         Unsane   
50964         Unsane     SE                0.0         Unsane   
50965         Unsane     IT                0.0         Unsane   
50966         Unsane     US                0.0         Unsane   

      original_title_x  ...  Unnamed: 0       genre_ids    id_y  \
0        Hail, Caesar!  ...       17497  [35, 18, 9648]  270487   
1        Hail, Caesar!  ...       17497  [35, 18, 9648]  270487   
2        Hail, Caesar!  ...       17497  [35, 18, 9648]  270487   
3        Hail, Caesar!  ...       17497  [35, 18, 9648]  270487   
4        Hail, Caesar!  ...       17497  [35, 18, 9648]  270487   
...                ...  ...         ...             ...     ...   
50962           Unsane  ...       23949        [27, 53]  467660   
50963           Unsane  ...       23949        [27, 53]  467660   
50964           Unsane  ...       23949        [27, 53]  467660   
50965           Unsane  ...       23949        [27, 53]  467660   
50966           Unsane  ...       23949        [27, 53]  467660   

       original_language  original_title_y popularity release_date_y  \
0                     en     Hail, Caesar!     12.312     2016-02-05   
1                     en     Hail, Caesar!     12.312     2016-02-05   
2                     en     Hail, Caesar!     12.312     2016-02-05   
3                     en     Hail, Caesar!     12.312     2016-02-05   
4                     en     Hail, Caesar!     12.312     2016-02-05   
...                  ...               ...        ...            ...   
50962                 en            Unsane     16.316     2018-03-23   
50963                 en            Unsane     16.316     2018-03-23   
50964                 en            Unsane     16.316     2018-03-23   
50965                 en            Unsane     16.316     2018-03-23   
50966                 en            Unsane     16.316     2018-03-23   

      vote_average  vote_count     profits  
0              5.9        2328  42160680.0  
1              5.9        2328  42160680.0  
2              5.9        2328  42160680.0  
3              5.9        2328  42160680.0  
4              5.9        2328  42160680.0  
...            ...         ...         ...  
50962          6.2         667  12744931.0  
50963          6.2         667  12744931.0  
50964          6.2         667  12744931.0  
50965          6.2         667  12744931.0  
50966          6.2         667  12744931.0  

[50967 rows x 37 columns]

In [89]:
film_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50967 entries, 0 to 50966
Data columns (total 37 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   movie_id            50967 non-null  object 
 1   ordering            50967 non-null  int64  
 2   person_id           50967 non-null  object 
 3   category            50967 non-null  object 
 4   ordering.1          50967 non-null  int64  
 5   title               50967 non-null  object 
 6   region              50967 non-null  object 
 7   is_original_title   50967 non-null  float64
 8   primary_title       50967 non-null  object 
 9   original_title_x    50967 non-null  object 
 10  start_year          50967 non-null  int64  
 11  runtime_minutes     50967 non-null  float64
 12  genres              50967 non-null  object 
 13  averagerating       50967 non-null  float64
 14  numvotes            50967 non-null  int64  
 15  primary_name        50967 non-null  object 
 16  prim

In [90]:
box_office_df = pd.read_csv('zippedData/bom.movie_gross.csv.gz')
box_office_df

title      studio  domestic_gross  \
0                                     Toy Story 3          BV     415000000.0   
1                      Alice in Wonderland (2010)          BV     334200000.0   
2     Harry Potter and the Deathly Hallows Part 1          WB     296000000.0   
3                                       Inception          WB     292600000.0   
4                             Shrek Forever After        P/DW     238700000.0   
...                                           ...         ...             ...   
3382                                    The Quake       Magn.          6200.0   
3383                  Edward II (2018 re-release)          FM          4800.0   
3384                                     El Pacto        Sony          2500.0   
3385                                     The Swan  Synergetic          2400.0   
3386                            An Actor Prepares       Grav.          1700.0   

     foreign_gross  year  
0        652000000  2010  
1        691300000  2010  
2        664300000  2010  
3        535700000  2010  
4        513900000  2010  
...            ...   ...  
3382           NaN  2018  
3383           NaN  2018  
3384           NaN  2018  
3385           NaN  2018  
3386           NaN  2018  

[3387 rows x 5 columns]

In [91]:
film_df['foreign_gross'].head()

0    33100000
1    33100000
2    33100000
3    33100000
4    33100000
Name: foreign_gross, dtype: object

In [92]:
#foreign gross was stored as a object we need to change it to a numeric

box_office_df['foreign_gross']=pd.to_numeric(box_office_df['foreign_gross'],errors='coerce')
box_office_df['revenue'] = box_office_df['domestic_gross'] + box_office_df['foreign_gross']



In [93]:
# Merge the two DataFrames first (assuming they both have 'film_id')
merged_df = pd.merge(film_df, box_office_df, on='title')

# Now you can filter easily:
action_movies_revenue = merged_df[merged_df['genres'] == 'Action,Adventure,Sci-Fi']['revenue']
animation_movies_revenue = merged_df[merged_df['genres'] == 'Adventure,Animation,Comedy']['revenue']


In [94]:
film_df['genres'].value_counts()

Action,Adventure,Sci-Fi        3020
Adventure,Animation,Comedy     2920
Drama,Romance                  1905
Action,Adventure,Drama         1860
Action,Adventure,Fantasy       1770
                               ... 
Fantasy,Horror                    7
Crime,Documentary,History         6
Action,Comedy,Drama               6
Musical                           5
Adventure,Documentary,Drama       4
Name: genres, Length: 204, dtype: int64

#### Filter the various `Genres and establish the sample data frame to be used`

In [95]:


# Filter for movies with the genre 'Action,Adventure,Sci-Fi'
filtered_df = film_df[film_df['genres'] == 'Adventure,Animation,Comedy']

# Get 30% of the filtered dataset as a random sample
sample_df = filtered_df.sample(frac=0.3, random_state=42)  # Use random_state for reproducibility

print(sample_df)

        movie_id  ordering  person_id  category  ordering.1            title  \
45574  tt3411444         6  nm0047916    writer          11        Ferdinand   
45607  tt3411444         8  nm0178589    writer          24        Ferdinand   
45496  tt3411444        10  nm0452235    writer          24        Ferdinand   
20206  tt1860353         1  nm0005351     actor          21            Turbo   
34719  tt2296777         2  nm0564215     actor           8  Sherlock Gnomes   
...          ...       ...        ...       ...         ...              ...   
7294   tt2294629         3  nm2676147     actor          44           Frozen   
32806  tt0435761         5  nm0881279  director          22      Toy Story 3   
20329  tt1860353         8  nm0829601  producer          27            Turbo   
15314  tt1436562        10  nm0893262    writer          20              Rio   
45515  tt3411444         2  nm0571952   actress           9        Ferdinand   

      region  is_original_title    prim

#### Created the samples that will be used in `Statistical Computation`

In [96]:

# Filter for Action, Adventure, Sci-Fi genre movies
action_adventure_sci_fi = film_df[film_df['genres'].str.contains('Action|Adventure|Sci-Fi', na=False)]

# Filter for Adventure Animation genre movies
animation_movies= film_df[film_df['genres'].str.contains('Adventure|Animation|Comedy', na=False)]

# Take a sample of 30% from each group
sample_action = action_adventure_sci_fi.sample(frac=0.3, random_state=42)
sample_animation = animation_movies.sample(frac=0.3, random_state=42)

print("Action/Adventure/Sci-Fi Sample Size:", len(sample_action))
print(" Adventure/Animation/Comedy Sample Size:", len(sample_animation))


Action/Adventure/Sci-Fi Sample Size: 8109
 Adventure/Animation/Comedy Sample Size: 7716


## STATISTICAL ANALYSIS

Now that we’ve created the samples, let’s `formulate the hypotheses`:
- Null Hypothesis (H₀): The average box office revenue for Action/Adventure/Sci-Fi movies is equal to that of Adventure/Animation /Comedy movies.
- Alternative Hypothesis (H₁): The average box office revenue for Action/Adventure/Sci-Fi movies is different (or higher) than Adventure/Animation/Comedy movies.


## Sample Data loaded lets now begin to : Perform T-Test
Create a revenue column by combinining the columns for domestic gross and foreign gross 


In [97]:
#creating a new column revenue for hypothesis testing
film_df['revenue'] = box_office_df['domestic_gross'] + box_office_df['foreign_gross']

Load the necessary libraries `import scipy.stats` to formulate the ttest

In [98]:
from scipy.stats import ttest_ind

# 1. Filter Action and Animation movies
sample_action = film_df[film_df['genres'] == 'Action/Adventure/Sci-Fi']
sample_animation= film_df[film_df['genres'] == 'Adventure,Animation,Comedy']

# 2. Extract revenue columns
action_revenue = sample_action['revenue']
animation_movies_revenue = sample_animation['revenue']

# 3. Perform Welch's t-test
t_stat, p_value = ttest_ind(action_revenue,animation_movies_revenue, equal_var=False)

# 4. Print Results
print("T-Statistic:", t_stat)
print("P-Value:", p_value)

# 5. Interpretation
if p_value <= 0.05:
    print("Reject the null hypothesis: Significant difference in revenue.")
else:
    print("Fail to reject the null hypothesis: No significant difference in revenue.")


T-Statistic: nan
P-Value: nan
Fail to reject the null hypothesis: No significant difference in revenue.


In [99]:
# Check for missing or invalid values in Revenue
print(sample_action['revenue'].isna().sum())  # Count missing values in the Action sample
print(sample_animation['revenue'].isna().sum())   # Count missing values in the Animation sample

# Drop rows with missing or invalid values in Revenue
sample_action = sample_action.dropna(subset=['revenue'])
sample_drama = sample_animation.dropna(subset=['revenue'])

0
2871


#### Calculating Confidence Interval 

In [100]:

# Calculate means and standard deviations
mean_action = np.mean(action_revenue)
std_action = np.std(action_revenue, ddof=1)
mean_animation = np.mean(animation_movies_revenue)
std_animation = np.std(animation_movies_revenue, ddof=1)

# Calculate confidence intervals (95%)
conf_interval_action = (mean_action - 1.96 * std_action/np.sqrt(len(action_revenue)),
                        mean_action + 1.96 * std_action/np.sqrt(len(action_revenue)))

conf_interval_animation = (mean_animation - 1.96 * std_animation/np.sqrt(len(animation_movies_revenue)),
                           mean_animation + 1.96 * std_animation/np.sqrt(len(animation_movies_revenue)))

print("Confidence Interval for Action/Adventure/Sci-Fi:", conf_interval_action)
print("Confidence Interval for Adventure/Animation/Comedy:", conf_interval_animation)


Confidence Interval for Action/Adventure/Sci-Fi: (nan, nan)
Confidence Interval for Adventure/Animation/Comedy: (419566983.4521813, 451430743.03761464)


Interpretation:
You are 95% confident that the true mean box office revenue for Adventure/Animation movies lies between approximately $419.57 million and $451.43 million.

What This Means in Practice:
Stakeholders can use this range for planning budgets and making investment decisions. Moreover, if you compare this interval with the one from another genre (like Action/Adventure/Sci-Fi), non-overlapping intervals would suggest a statistically significant difference between the two groups


### Debug the data in Action /Adventure /Sci-Fi column filter the data and clean for Missing values and convert the revenue column to numeric value
 The columns Action/Adventure/Sci-Fi contain Null values or Missing values and drop them.
 The filtering the sample data and assertain whether the data in revenue is numeric value.format And ensure the datatype is correct.

In [101]:
# Convert Revenue to numeric in the main DataFrame if not already done
film_df['revenue'] = pd.to_numeric(film_df['revenue'], errors='coerce')

# Redefine your filtered sample for Action/Adventure/Sci-Fi
action_adventure_sci_fi = film_df[film_df['genres'].str.contains('Action|Adventure|Sci-Fi', na=False)]

# Sample 30% (make sure the sample is not empty)
sample_action = action_adventure_sci_fi.sample(frac=0.3, random_state=42)

# Drop missing revenue values
sample_action = sample_action.dropna(subset=['revenue'])

print("Updated Sample Action shape:", sample_action.shape)
print(sample_action['revenue'].describe())


Updated Sample Action shape: (228, 38)
count    2.280000e+02
mean     1.092280e+08
std      2.025094e+08
min      4.900000e+03
25%      4.825000e+06
50%      2.475000e+07
75%      1.136718e+08
max      1.153300e+09
Name: revenue, dtype: float64


### Perform a data cleaning for computation of the Confidence interval Once again

In [102]:

# Ensure the 'Revenue' column is numeric and drop any rows where Revenue is missing
film_df['revenue'] = pd.to_numeric(film_df['revenue'], errors='coerce')
film_df = film_df.dropna(subset=['revenue'])

# Filter for Adventure/Animation movies (adjust genre filtering as needed)
sample_animation = film_df[film_df['genres'].str.contains("Adventure|Animation", na=False)]

# Optional: Take a 30% random sample from the filtered Adventure/Animation movies
sample_animation = sample_animation.sample(frac=0.3, random_state=42)

# After sampling, drop rows with missing 'Revenue', just in case
sample_animation = sample_animation.dropna(subset=['revenue'])

# Calculate mean, standard deviation, and sample size for the Adventure/Animation sample
mean_animation = np.mean(sample_animation['revenue'])
std_animation = np.std(sample_animation['revenue'], ddof=1)  # Sample standard deviation
n_animation = len(sample_animation['revenue'])

# Print summary statistics
print("Mean Revenue (Adventure/Animation/Comedy):", mean_animation)
print("Standard Deviation (Adventure/Animation/Comedy):", std_animation)
print("Sample Size (Adventure/Animation/Comedy):", n_animation)

# Calculate the 95% confidence interval if the sample is not empty
if n_animation > 0:
    ci_animation = (mean_animation - 1.96 * std_animation / np.sqrt(n_animation),
                    mean_animation + 1.96 * std_animation / np.sqrt(n_animation))
    print("Confidence Interval for Adventure/Animation/Comedy:", ci_animation)
else:
    print("No valid revenue data in Adventure/Animation/Comedy sample to compute confidence interval.")


Mean Revenue (Adventure/Animation/Comedy): 109915290.37179486
Standard Deviation (Adventure/Animation/Comedy): 213613537.05051163
Sample Size (Adventure/Animation/Comedy): 156
Confidence Interval for Adventure/Animation/Comedy: (76393859.88243532, 143436720.8611544)


Both confidence intervals provide valuable insights, and the choice isn’t clear‑cut solely by these numbers because they overlap substantially. However, we can explore the details to help investors decide based on their risk tolerance and business objectives. Here’s the interpretation:
The Numbers You Have
- Action/Adventure/Sci‑Fi Confidence Interval:
($82.94 \text{M}, $135.51 \text{M})
- Adventure/Animation Confidence Interval:
($76.39 \text{M}, $143.44 \text{M})

What These Tell Us
- Average Revenue (Point Estimate):
Although the point estimates (means) for both genres are very similar—around $109–110 million—the confidence interval for each shows the range in which the true mean revenue is expected to lie with 95% confidence.
- Lower Bound Comparison:- Action/Adventure/Sci‑Fi: $82.94M
- Adventure/Animation: $76.39M
The higher lower bound in Action/Adventure/Sci‑Fi suggests that, at worst, these movies tend to generate a bit more revenue on average. This can be attractive if investors value consistency and a lower risk of underperforming.
- Upper Bound Comparison:- Action/Adventure/Sci‑Fi: $135.51M
- Adventure/Animation: $143.44M
The Adventure/Animation group has a higher upper bound, hinting at the possibility of achieving larger blockbuster-like revenues. This indicates higher potential upside, but it comes with more variability.
- Variability:
The wider confidence interval for Adventure/Animation implies increased variability. In practical terms, there might be more fluctuation in revenue outcomes within this genre, meaning a higher risk–reward scenario.

The Action Adventure Science Fiction is our Genre to Pitch to Investors
- If Investors Prefer Consistency and Lower Risk:
Pitch Action/Adventure/Sci‑Fi. Its slightly higher lower bound and narrower interval suggest a more consistent revenue performance, providing a more predictable return on investment.
- If Investors Are Willing to Embrace Risk for a Higher Upside:
Pitch Adventure/Animation. The higher upper bound indicates that, while there is greater variability, there’s also the opportunity for a bigger blockbuster hit. This may appeal to investors who are comfortable with some risk in exchange for the chance at higher rewards.

Final Recommendation 

Because the intervals overlap substantially, statistically, there isn’t a dramatic difference in the mean revenue expectation between the two groups. Your decision might then hinge on investors' overall risk appetite and strategic objectives:
- Conservative (Steady Returns): Action/Adventure/Sci‑Fi
- Aggressive (High Upside Potential): Adventure/Animation

In summary, if our investors lean towards predictability, the Action/Adventure/Sci‑Fi genre would be the best to produce based on these confidence intervals. However, if they’re seeking transformational, blockbuster returns and are comfortable with higher variance, then Adventure/Animation might be more exciting.
Ultimately, you might also consider other factors like market trends, cost of production, and potential audiences.


### 2 .Hypothesis Formulation

#### B.Impact of Budget on Box Office Revenue
`To evaluate whether investments on production budgets significantly affect box office performance`
- Null Hypothesis (H₀): Budget has no significant impact on box office revenue.
- Alternative Hypothesis (H₁): Higher budgets lead to significantly higher box office revenue

In [103]:
#import library
import pandas as pd
#load the csv
budget_df=pd.read_csv('zippedData\cleaned_dataset_movie_budget.csv')
budget_df.head()

id  release_date                                        movie  \
0   1  dec 18, 2009                                       avatar   
1   2  may 20, 2011  pirates of the caribbean: on stranger tides   
2   3   jun 7, 2019                                 dark phoenix   
3   4   may 1, 2015                      avengers: age of ultron   
4   5  dec 15, 2017            star wars ep. viii: the last jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [104]:
budget_df.columns

Index(['id', 'release_date', 'movie', 'production_budget', 'domestic_gross',
       'worldwide_gross'],
      dtype='object')

In [105]:
#rename worlwide_gross to revenue
budget_df=budget_df.rename(columns={'worldwide_gross':'revenue'})
print (budget_df.columns)

Index(['id', 'release_date', 'movie', 'production_budget', 'domestic_gross',
       'revenue'],
      dtype='object')


In [106]:
#rename production_budget to budget
budget_df=budget_df.rename(columns={'production_budget':'budget'})

Proceed to check for missing values

In [107]:
# Remove non-numeric characters (commas, dollar signs) and convert to numeric
for col in ['budget', 'revenue']:
    budget_df[col] = budget_df[col].astype(str).str.replace(r'[^\d.]', '', regex=True)
    budget_df[col] = pd.to_numeric(budget_df[col], errors='coerce')


In [108]:
#drop rows with missing values
budget_df = budget_df.dropna(subset=['budget', 'revenue'])


Finalise the data cleaning .Assertaining the budget column has no Nan values

In [109]:
# Force all to strings and lowercase
budget_df['budget'] = budget_df['budget'].astype(str).str.lower()

# Replace common bad values with NaN
budget_df['budget'] = budget_df['budget'].replace(['n/a', 'na', '-', 'none', 'unknown', '—'], pd.NA)

# Remove all non-numeric characters (except decimal point)
budget_df['budget'] = budget_df['budget'].str.replace(r'[^\d.]', '', regex=True)

# Convert to numeric
budget_df['budget'] = pd.to_numeric(budget_df['budget'], errors='coerce')




 We will need to obtain High budget and the Low budget threshhold by computation of the median production.This will split by the median 

In [110]:
#COMPUTE THE MEDIAN PRODUCTION FOR THE BUDGET ANALYSIS
# Find the median budget
median_budget = budget_df['budget'].median()
print("Median budget:", median_budget)

# Split into high and low budget groups
low_budget_df = budget_df[budget_df['budget'] < median_budget]
high_budget_df = budget_df[budget_df['budget'] >= median_budget]



Median budget: 17000000.0


 Median Production Budget Analysis
The dataset was cleaned to extract valid numerical values for both budget and revenue. After removing any non-numeric characters and null entries, we computed the median production budget across all films.

Median Budget: $17,000,000

This median value serves as the threshold to divide the dataset into two groups for further statistical analysis:

Low Budget Films: Films with budgets ≤ $17M

High Budget Films: Films with budgets > $17M

This grouping is essential for hypothesis testing to determine whether higher production budgets are associated with significantly higher box office revenues.
We have assertained that the median value of 17,000,000 will be the threshhold of our statistical analysis.
Let us now calculate the T-statistic and P-value

In [111]:
from scipy.stats import ttest_ind

# Extract revenue
low_revenue = low_budget_df['revenue']
high_revenue = high_budget_df['revenue']

# Run Welch’s t-test
t_stat, p_value = ttest_ind(high_revenue, low_revenue, equal_var=False)

# Display results
print("T-Statistic:", t_stat)
print("P-Value:", p_value)

# Interpretation
if p_value <= 0.05:
    print("Reject H₀: High-budget films earn significantly more revenue.")
else:
    print("Fail to reject H₀: No significant revenue difference based on budget.")


T-Statistic: 32.12504153761319
P-Value: 9.18893427925073e-197
Reject H₀: High-budget films earn significantly more revenue.



` The Hypothesis Test: Impact of Budget on Box Office Revenue
To evaluate whether production budgets significantly affect box office performance, a two-sample t-test was conducted comparing revenues from:`

High Budget Films with budgets greater than the 17,000,000 threshold (budget > $17M)

Low Budget Films equal or lesser than the 17,000,000 threshold .(budget ≤ $17M)

 Hypotheses 
Null Hypothesis (H₀): Budget has no significant impact on box office revenue.

Alternative Hypothesis (H₁): Higher budgets lead to significantly higher revenue.

 Test Details
Test Used: Welch’s t-test (accounts for unequal variances)

T-statistic: 32.13

P-value: 9.19 × 10⁻¹⁹⁷

`We thereby Reject the null hypothesis.`

There is overwhelming statistical evidence that higher-budget films earn significantly more revenue than lower-budget films.

In [112]:
#COMPUTING THE CONFIDENCE INTERVALS
# Ensure Revenue is numeric
budget_df['revenue'] = pd.to_numeric(budget_df['revenue'], errors='coerce')
budget_df = budget_df.dropna(subset=['revenue'])

# Compute mean, standard deviation, and sample size for low-budget movies
mean_low = np.mean(low_budget_df['revenue'])
std_low = np.std(low_budget_df['revenue'], ddof=1)
n_low = len(low_budget_df['revenue'])

# Compute mean, standard deviation, and sample size for high-budget movies
mean_high = np.mean(high_budget_df['revenue'])
std_high = np.std(high_budget_df['revenue'], ddof=1)
n_high = len(high_budget_df['revenue'])

# Calculate confidence intervals for both groups
if n_low > 0:
    ci_low = (mean_low - 1.96 * std_low / np.sqrt(n_low), 
              mean_low + 1.96 * std_low / np.sqrt(n_low))
    print("Confidence Interval for Low-Budget Movies:", ci_low)
else:
    print("Not enough data for Low-Budget CI.")

if n_high > 0:
    ci_high = (mean_high - 1.96 * std_high / np.sqrt(n_high), 
               mean_high + 1.96 * std_high / np.sqrt(n_high))
    print("Confidence Interval for High-Budget Movies:", ci_high)
else:
    print("Not enough data for High-Budget CI.")

Confidence Interval for Low-Budget Movies: (21468885.009598993, 25226213.171140715)
Confidence Interval for High-Budget Movies: (150432556.5062074, 166485429.27431387)
